### ISSUE TITLE
[$2000] When you long-press on a message in iOS mobile, the message moves out of the screen frame and is hidden

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#10632]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to any chat with multiple messages 
2. Long press on the last message you sent

## Expected Result:
When long pressing on the last message, we should adjust the scroll so that the message you just selected is visible. 

## Actual Result:
When you long-press on a message in iOS mobile, the message moves out of the screen frame and is hidden.

## Workaround:
None 

## Platform:

Where is this issue occurring?

- iOS

**Version Number:** 1.1.92-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43995119/187123528-d0427883-8364-40be-899d-4ba63c33d98d.MP4

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @roryabraham 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1661612352759039

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by marcaaron** [comment]
>User should be able to see which messages is being selected

What does this mean exactly... ? Looking at the video and I'm not even sure what the bug is.

**Comment by marcaaron** [comment]
>When you long-press on a message in iOS mobile, the message moves out of the screen frame, and it’s not obvious which message you’ve selected

Also don't understand what is being proposed here.

**Comment by roryabraham** [comment]
The problem is that once the context menu pops up and the modal blur is applied, it's not clear which message you've long-pressed on. What I was imagining is that the message you selected would have a white background / be "immune" from the modal blur effect.

But I think a mockup would be very helpful here. @marcaaron suggested that we could use a "shimmer effect" as an alternative

**Comment by michelle-thompson** [comment]
Hmm, what about when you hold down on a message the background highlights as grey like so?
![image](https://user-images.githubusercontent.com/43361905/189102765-d1a24f6b-a628-4634-a090-dacb6fb68fa8.png)

**Comment by roryabraham** [comment]
Yeah, that works! We'll also need to make sure that we:

- Scroll the chat view so that the highlighted chat is visible despite the context menu opening
- Make sure we un-highlight the option when the context menu closes

**Comment by roryabraham** [comment]
Leaving you assigned so that you get the GH points @michelle-thompson

**Comment by roryabraham** [comment]
Sending this one to be external, keeping it a weekly for now

**Comment by JmillsExpensify** [comment]
Catching up on the above before exporting.

**Comment by JmillsExpensify** [comment]
Alright, had to think on this one a bit. I'm going to export it because I think it fits squarely with WhatsApp quality, specifically polish of existing functionality. (As a general rule, I think we should avoid adding any explicit new functionality while WhatsApp quality is in effect).

**Comment by JmillsExpensify** [comment]
Job created here: https://www.upwork.com/jobs/~01be1724c1619fb8dc

**Comment by JmillsExpensify** [comment]
Still open for proposals

**Comment by JmillsExpensify** [comment]
Still waiting for proposals.

**Comment by JmillsExpensify** [comment]
Doubling the price to encourage proposals.

**Comment by Santhosh-Sellavel** [comment]
Thanks, @mollfpr.

@roryabraham @michelle-thompson 
Seems the highlight is already available, it's not visible clearly as @mollfpr said.

> >Scroll the chat view so that the highlighted chat is visible despite the context menu opening
> 
> If this means that we want to open the context menu of the last chat, we should make the chat visible rather than got covered by the context menu?

Interesting how are we gonna show context for this case?

**Comment by roryabraham** [comment]
> If this means that we want to open the context menu of the last chat, we should make the chat visible rather than got covered by the context menu?

Correct. Good point – it might not be so much scrolling as pushing the whole `ReportActionsView` up to keep it in view. But let's imagine we make that change, and then you long-press on the top chat. Pushing it up would push it out of view, so you'd actually have to scroll up to keep it visible.

> Seems the highlight is already available, it's not visible clearly as @mollfpr said.

Good point, I didn't notice this. Almost seems like we'd want the blurred overlay to happen everywhere _except_ for the item you're highlighting. cc @michelle-thompson 

Also seems like this could become a very complicated change ... If the bottom-docked menu pushes content up out of its way, then it's no longer a modal.

**Comment by Santhosh-Sellavel** [comment]
> Also seems like this could become a very complicated change ... If the bottom-docked menu pushes content up out of its way, then it's no longer modal.

Yeah but this is how WhatsApp handles long presses on iOS
![IMG_CFD18AAEAA44-1](https://user-images.githubusercontent.com/85645967/195909264-5f519d58-2b62-4598-aefc-ab0889ba87cf.jpeg)

Just like Messages in iOS, blur out other messages.
https://user-images.githubusercontent.com/85645967/195909601-3c32273f-c9e0-450c-b176-41b7e9d667ea.MP4

But android it's a different story, they provide context options in the header

**Comment by JmillsExpensify** [comment]
It doesn't look like we've made much movement on this one in the last week. @roryabraham @Santhosh-Sellavel - I'm going to double the price of this issue as a result. 

That said, I also want to come back to this comment from @roryabraham. 

> Also seems like this could become a very complicated change ... If the bottom-docked menu pushes content up out of its way, then it's no longer a modal.

Let's align on our best approach. Perhaps we should more appropriately pause this issue, or at the very least, consider this a new feature / polish rather than a bug. If that's the case, I'd argue we should take this back to Slack and align on the intended behavior before moving forward with said feature/polish. 

Thoughts?

**Comment by Santhosh-Sellavel** [comment]
Still looking for proposals!

**Comment by JmillsExpensify** [comment]
Before we circle back on proposals, I think we should align on my last comment.

**Comment by Santhosh-Sellavel** [comment]
Let's do that @JmillsExpensify 
Remove `Help Wanted`, 'Bug' labels.
Please apply the ` New Feature` Label, and also apply the `design` label!

**Comment by Santhosh-Sellavel** [comment]
I just saw @shawnborton  is already assigned here!

**Comment by JmillsExpensify** [comment]
Cool, I'm also putting this on hold given that it's not a roadmap priority.

**Comment by roryabraham** [comment]
Sorry for the delayed response here, but I personally don't agree that this is a new feature, I think it's a bug

**Comment by JmillsExpensify** [comment]
Quick update on this thread, as I just got on a call with @roryabraham and we aligned on what's happened here. Basically this comes down to two separate issues:

 - **Bug**: If you long tap on the last message you sent, then the bottom drawer completely covers the message you just selected. 
 - **Polish**: If you long tap on any message, it's not very clear which message you selected. This is because the highlight is extremely subtle, and perhaps you could argue we don't pop/animate the selection either. 

So we should align on what the best next steps are now that this is hopefully clear to all, or minimally to me. 😄

**Comment by JmillsExpensify** [comment]
I tend to think we should solve the bug first and foremost. However, in solving the bug I could also see the argument that we should also solve the highlight too.

**Comment by roryabraham** [comment]
Since we agree that there is a bug here, should we remove the `[HOLD]` and switch it from `New Feature` to `Bug`?

**Comment by roryabraham** [comment]
Also, maybe we should separate the two concerns into separate issues?

**Comment by JmillsExpensify** [comment]
I'm not super passionate about whether we solve these separately or together. If we're just improving the color we use to highlight the message, then we could add both in one and raise the price. Does anyone have a strong preference one way or the other?

**Comment by JmillsExpensify** [comment]
Went ahead and removed the hold! I also updated this issue with a more clear description of the bug. I'll circle back on the polish issue, not a priority at this very moment.

**Comment by Santhosh-Sellavel** [comment]
Recently came off hold, open for proposals or not @JmillsExpensify?

**Comment by JmillsExpensify** [comment]
Yes! We're open for proposals. Upwork job is here: https://www.upwork.com/jobs/~01be1724c1619fb8dc. Kicking is off at the last price, which is also the new starting price for bugs.

**Comment by Santhosh-Sellavel** [comment]
not overdue, looking fir proposals

**Comment by JmillsExpensify** [comment]
Still waiting on proposals. I'll swing back for a price increase next week.

**Comment by roryabraham** [comment]
Still waiting for proposals, brought up in slack [here](https://expensify.slack.com/archives/C035J5C9FAP/p1669710641134479?thread_ts=1669655465.254259&cid=C035J5C9FAP)

**Comment by JmillsExpensify** [comment]
Doubling the price in the meantime.

**Comment by JmillsExpensify** [comment]
Note: The previous upwork job expired, so the new one is here. https://www.upwork.com/jobs/~01c649adebe4e80002

**Comment by arosiclair** [comment]
I can take this internal if you guys don't mind @roryabraham

**Comment by Santhosh-Sellavel** [comment]
I believe it will be 👍 only @arosiclair, please take it. Remove the `Help wanted` label and assign yourself. Keep me assigned I'll help with the PR review, thanks!

**Comment by roryabraham** [comment]
Thanks @arosiclair, I'd like to review the PR when it's ready

**Comment by arosiclair** [comment]
I repro'd this in dev on iOS, Android and mWeb. 

After thinking about this a bit more I'd have to echo Rory's sentiment about the complexity of fixing this. The context menu is a modal so it intentionally covers the conversation view. We could try scrolling the selected message up to the top so it doesn't get covered, but it would still be blurred and messages at the bottom of the conversation have no more room to scroll. 

I think the only true way to fix this is to add a preview of the selected message to the menu. Something along the lines of this:

<img src="https://user-images.githubusercontent.com/6833644/204925005-181b357c-e6d7-4133-bacb-6841cc8ea10e.png" height="300">

Adding that is squarely new feature/polish territory though and maybe could use some design input. What do you think @JmillsExpensify?

**Comment by JmillsExpensify** [comment]
Ah that's interesting. That's one of the potential solutiosn that @roryabraham and I discussed over the phone when we aligned earlier. That would also put us back where we were before, which to echo you is `New Feature` territory. At that point I think we need to put this feature on HOLD because it's not a WAQ priority.

**Comment by JmillsExpensify** [comment]
@roryabraham As the issue reporter, I get the sense that you're pretty passionate about this one. Given that we're struggled with bug/feature a couple of times in this issue – and more recently, might fix the "bug" by building a new feature – I'm back to feeling that this isn't really a priority right now. Curious for your thoughts!

**Comment by roryabraham** [comment]
I unfortunately am not a fan of @arosiclair's proposed solution. I think that ultimately the solution to this is a big lift which changes the context menu such that it's no longer a modal but a menu that slides in from the bottom of the screen and pushes content up out of its way. It sucks that it's a big lift, but just because it's hard to do doesn't make it less of a priority. I see this as a glaring usability problem with the app so definitely wouldn't deprioritize

**Comment by arosiclair** [comment]
By complexity I meant design wise not so much implementation. Admittedly I skimmed and picked up this GH thinking it'd be a straight forward bug to fix but finding this to be more of a polish issue that could use more thought or a rework. With that in mind I'll unassign and free this up. 

FWIW I think either of our proposal's are worth a design discussion somewhere. The only concerns I'd have about yours is fitting the message on screen and performance. I get the feeling a big layout shift could end up janky on mid-lower end devices but I don't have any proof of that.

**Comment by Santhosh-Sellavel** [comment]
on hold 
@JmillsExpensify can we bump this down to weekly will we wait?

**Comment by JmillsExpensify** [comment]
Just came across this in my weekly sweep. Yeah agreed, removed daily and added weekly.

**Comment by JmillsExpensify** [comment]
Based on a discussion with @roryabraham, and recognizing that WAQ is nearing the finish line, it would be good to pick this issue back up. There is still an outstanding question around how exactly to do that, though regardless, he's right that this is worth finding a way to address this behavior. Anyone have any immediate ideas? I know @roryabraham mentioned potentially doing a mini-design doc, though we would still need a volunteer in any case and @arosiclair already has a lot on his hands with the notifications project.

**Comment by roryabraham** [comment]
Marc from Margelo suggested an alternative solution – using https://github.com/mpiannucci/react-native-context-menu-view to display a native context menu rather than a bottom-docked modal for the context menu.

**Comment by JmillsExpensify** [comment]
Nice one. Perhaps we should have Margelo take this one on now that WAQ is officially done?

**Comment by JmillsExpensify** [comment]
We're still looking for a volunteer on this one.

**Comment by JmillsExpensify** [comment]
@roryabraham let's come back to this one. Above we said that we'd keep it internal, though I wonder if this would be good for Callstack to take on? Let me know if you agree and I can float to them.

**Comment by JmillsExpensify** [comment]
@roryabraham let's come back to this one. Above we said that we'd keep it internal, though I wonder if this would be good for Margelo to take on? Let me know if you agree and I can float to them.

**Comment by JmillsExpensify** [comment]
Actually yeah. Discussed with Margelo, they're going to take it on! We'll have someone to assign soon.

**Comment by roryabraham** [comment]
👍🏼 to giving this to Margelo. Happy to sync with someone on their team on slack to chat about this further.

**Comment by terrysahaidak** [comment]
Terry from Margelo here. Taking this ticket.

**Comment by MelvinBot** [comment]
📣 @terrysahaidak! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by JmillsExpensify** [comment]
Nice one @terrysahaidak! I went ahead and assigned you.

**Comment by roryabraham** [comment]
Looking forward to reviewing the PR when it's ready

**Comment by terrysahaidak** [comment]
> Looking forward to reviewing the PR when it's ready

Hope today. One last bug to fix. When you open the keyboard and then long press on the message and pick the emoji picker, something triggers dismiss on the keyboard after the action sheet is hidden. React Native modal on iOS handles it in a way that the keyboard is hidden/shown without animation, immediately. You can see this behavior for the emoji picker itself, when the keyboard is open and you click the emoji picker icon either on the message or in the composer. But in this particular case with two models one by one something somewhere triggers dismiss with animation.

**Comment by terrysahaidak** [comment]
@roryabraham I'm checking out all the checkboxes in the PR preparing process, adding screen recordings, etc., but the code is ready for review. 

Perhaps, you could help me put some things into proper folders and, just in general, point to some best practices used in this repo I'm not aware of.

**Comment by MelvinBot** [comment]
This issue has not been updated in over ***15*** days. @JmillsExpensify, @shawnborton, @terrysahaidak, @Santhosh-Sellavel eroding to ***Monthly*** issue.

P.S. Is everyone reading this sure this is really a near-term priority? Be brave: if you disagree, go ahead and close it out. If someone disagrees, they'll reopen it, and if they don't: one less thing to do!

**Comment by terrysahaidak** [comment]
@JmillsExpensify Reanimated team recently released a new version of Reanimated with fixed required for the PR to get out of "on hold" state. In order to unblock this PR, we first need to actually upgrade Reanimated to the new version with the fixes, I'll do it tomorrow, it includes lots of other fixes so we should do it anyway.

There is another tiny flicker i noticed recently related to `useAnimatedKeyboard` Reanimated team was refactoring, but it could be a bug in React Native's Modal.

And another piece of this puzzle is KeyboardAvoidingView implemented using `useAnimatedKeyboard` - https://github.com/software-mansion/react-native-reanimated/pull/4339 - in my PR I was using a lightweight implementation of this, but it might not work in all the cases and this PR implements it with the same behavior (bugs/features/quirks) as the original implementation, so it can be a drop-in replacement but now we can react to keyboard avoiding view changes. The KeyboardAvoidingView PR has a tiny bug aside from the flicker I mentioned, plan to get it fixed yesterday too.

**Comment by JmillsExpensify** [comment]
>Reanimated team recently released a new version of Reanimated with fixed required for the PR to get out of "on hold" state. In order to unblock this PR, we first need to actually upgrade Reanimated to the new version with the fixes, I'll do it tomorrow, it includes lots of other fixes so we should do it anyway.

Oh, this is great. Thanks for clearing the next steps up and don't hesitate to bump us in Slack if we are blocking you in any way.

**Comment by chrispader** [comment]
Hey, sorry guys for the delay! Going to continue working on this now over [here](https://github.com/Expensify/App/pull/16356) 👍

**Comment by JmillsExpensify** [comment]
I believe that we're no longer prioritizing this PR in favor of other initiatives. That makes sense to me.

**Comment by chrispader** [comment]
Ok, @JmillsExpensify should we still work on it though when we have the time for it?

**Comment by JmillsExpensify** [comment]
Time permitting, go for it. I still think this has low value.

**Comment by hannojg** [comment]
Hey, @kirillzyusko just joined Margelo, and he is a perfect fit for this issue as he is a keyboard handling wizard!
Can you please assign this issue to him @JmillsExpensify ? Thanks

**Comment by kirillzyusko** [comment]
@hannojg @JmillsExpensify I think this problem can be fixed using `react-native-keyboard-controller` and its components (`KeyboardAvoidingView` etc.). I will try to do it 👍

**Comment by JmillsExpensify** [comment]
Whoops, sorry missed this one. Sound great!

**Comment by JmillsExpensify** [comment]
Waiting to be picked up

**Comment by kirillzyusko** [comment]
Contributor details
Your Expensify account email: kiryl@margelo.io
Upwork Profile Link:

**Comment by kirillzyusko** [comment]
This is being worked on in this PR: https://github.com/Expensify/App/pull/16356

**Comment by JmillsExpensify** [comment]
This has since moved to this PR: https://github.com/Expensify/App/pull/42143

**Comment by hannojg** [comment]
After switching to the new arch we've encountered a bug in reanimated on the fabric renderer.
We are currently trying to solve it to move on with this issue:

[Discussed here](https://expensify.slack.com/archives/C05LX9D6E07/p1718203093258459)

**Comment by perunt** [comment]
Hey @shawnborton, I left [a message](https://expensify.slack.com/archives/C035J5C9FAP/p1725008797544619?thread_ts=1724940987.149119&cid=C035J5C9FAP) here to clarify this PR. I look forward to your answer

**Comment by luacmartins** [comment]
Assigning myself to cover for Rory while he's on parental leave

**Comment by luacmartins** [comment]
Damn, it's a shame that we reverted the PR 😞 @kirillzyusko let's work on v2 taking the blockers into consideration

**Comment by JmillsExpensify** [comment]
Payment summary: @terrysahaidak $250 via Upwork.

**Comment by JmillsExpensify** [comment]
Actually wait a minute, there is a lot going on in this issue. @luacmartins It looks like there was at least one regression, so should the correct amount be $125?

**Comment by luacmartins** [comment]
This issue is not done yet. We reverted the [original PR](https://github.com/Expensify/App/pull/42143) because of a couple of regressions:

- https://github.com/Expensify/App/issues/54712
- https://github.com/Expensify/App/issues/54713

We're now working on v2 of the PR:

- https://github.com/Expensify/App/pull/54764

**Comment by JmillsExpensify** [comment]
Ah ok, gotcha. Also disregard the mention of $125 above. I misread the issue title. For now, I'll keep waiting on payment.

**Comment by luacmartins** [comment]
@perunt will be pushing the PR forward https://github.com/Expensify/App/pull/54764#issuecomment-2592540375

**Comment by JmillsExpensify** [comment]
Same as last week.

**Comment by luacmartins** [comment]
@perunt what's the ETA on this one?

**Comment by JmillsExpensify** [comment]
Checking in on that ETA? This has been years in the making, let's get it done and dusted.

**Comment by roryabraham** [comment]
Rejoining the fray as a reviewer now that I'm back from parental leave. I have ~2 years of context on this issue, so it seems fitting that I should help review the PR

**Comment by luacmartins** [comment]
We're still waiting on an ETA for the PR though. @perunt any updates? Let's get this one through the finish line

**Comment by perunt** [comment]
Sorry for the confusion. There was no availability for this task earlier, but I've just switched to it and made some adjustments. It should be good to go

**Comment by mallenexpensify** [comment]
Added to `#quality` as HIGH. The issue below is held on this
- https://github.com/Expensify/App/issues/37203

**Comment by JmillsExpensify** [comment]
PR in review

**Comment by mallenexpensify** [comment]
Is this the RP we're held on here?
- https://github.com/Expensify/App/pull/54764#issuecomment-2740768820

**Comment by JmillsExpensify** [comment]
I think so?

**Comment by JmillsExpensify** [comment]
No way a 3 year old issue is daily so I updated the label.

**Comment by JmillsExpensify** [comment]
Payment summary: $250 for @terrysahaidak for PR review and testing.

**Comment by JmillsExpensify** [comment]
Actually wait a second. This is a super long issue @roryabraham can you confirm where we ended up for the payment. This is a nearly three year old issue.

**Comment by perunt** [comment]
> Actually wait a second. This is a super long issue @roryabraham can you confirm where we ended up for the payment. This is a nearly three year old issue.

@terrysahaidak is a part of the Margelo team, so no payments needed

**Comment by JmillsExpensify** [comment]
Ah ok, thanks for the clarification. @rojiphil I see your reviewer checklist on the most recent PR. It looks like that's the PR that ultimately fixed this issue.

**Comment by rojiphil** [comment]
> I see your reviewer checklist on the most recent PR. It looks like that's the PR that ultimately fixed this issue.

Yeah, that’s correct @JmillsExpensify and the payment summary is due for C+ role here. Thanks for bringing this up.

With fixing related issues and working on two versions due to an unfortunate revert, the PRs has a story to tell of its own. 

As C+, please note that @shubham1206agra also worked on the [first PR](https://github.com/Expensify/App/pull/42143) whereas I picked it up [here](https://github.com/Expensify/App/pull/42143#issuecomment-2484692576) and navigated it through the [second PR](https://github.com/Expensify/App/pull/54764). And I am not sure of the process for compensation here, as the initial compensation of $2000 was also reset around [here](https://github.com/Expensify/App/issues/10632#event-16542402468)   
cc @luacmartins, @mallenexpensify, @mountiny

**Comment by shubham1206agra** [comment]
@JmillsExpensify I don't think the migration is over yet.

See https://github.com/Expensify/App/issues/37203

**Comment by terrysahaidak** [comment]
it's cool to see this issue being finally resolved. i did the initial implementation back then that basically remained the same. But big thanks to @kirillzyusko in general and his react-native-keyboard-controller, and I guess to all the others who contributed in react native over that time that finally resolved all the random synchronisation issues that would result in flickers when opening/closing  keyboard and/or modal screens.

**Comment by rojiphil** [comment]
> I don't think the migration is over yet.

Agree that the migration is not over yet. But compensation seems to be for each issue separately as per [this comment](https://github.com/Expensify/App/issues/37203#issuecomment-1964599278). I think it makes sense too as it is simpler to address then and there. cc @JmillsExpensify

**Comment by JmillsExpensify** [comment]
Ok, so given that the issue is still not done, we'll keep this one open. Equally @rojiphil has successfully completed the work assigned via the merged PR that has successfully gone through the regression period, so I will handle that now. 

Summary: $250 to @rojiphil for PR review and testing.

**Comment by JmillsExpensify** [comment]
Or alternatively, @rojiphil if you want to wait until everything is complete, then we can figure out the relative effort compared to the final result, whenever this is finally delivered.

**Comment by shubham1206agra** [comment]
@JmillsExpensify, is there a reason why I am not being paid here? Also, why is there a reduction in the amount?

**Comment by rojiphil** [comment]
@JmillsExpensify Thanks. My preference would be to handle the compensation then and there as that is simpler to handle. But always happy to pick up any further PRs for review as and when it comes up. 

> Summary: $250 to [@rojiphil](https://github.com/rojiphil) for PR review and testing.

Regarding the compensation, I believe it was more work than what we regularly do for the default compensation. The reasoning being that while the scope of changes that required review and testing itself was large, there were also quite a lot of review cycles we went through in 2 PRs. I believe the nature of work involved here demanded that and I think this deserves a raise in compensation. 
$500 seems reasonable. What do you think?

**Comment by JmillsExpensify** [comment]
@shubham1206agra no one is being paid yet. The issue is still outstanding.

**Comment by JmillsExpensify** [comment]
Cool, I'll post internally in Slack so we can align before wrapping this up.

**Comment by rojiphil** [comment]
paid
